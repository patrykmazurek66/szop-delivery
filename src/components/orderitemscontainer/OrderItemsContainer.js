import OrderItem from "../orderitem/OrderItem";
import { useState, useEffect } from "react";
import "./orderitemscontainer.css";
import Div from "../div/Div";
import P from "../p/P";

export default function OrderItemsContainer() {
  const [cart, setCart] = useState([]);
  const [items, setItems] = useState([]);
  const [cost, setCost] = useState(0);

  function refresh() {
    setCart(JSON.parse(localStorage.cart));
  }

  useEffect(() => {
    setCart(JSON.parse(localStorage.cart));
  }, []);

  useEffect(() => {
    setItems(
      cart.map(item => {
        return (
          <OrderItem
            name={item.name}
            price={item.price}
            qty={item.quantity}
            key={item.name}
            refresh={refresh}
          />
        );
      })
    );
    setCost(
      cart.reduce((acc, el) => {
        return (acc = acc + el.price * el.quantity);
      }, 0)
    );
  }, [cart]);

  return (
    <Div outline column nomp>
      <Div full nomp>
        <P header>Your Order</P>
      </Div>

      <Div column nomp full>
        {items}
      </Div>
      <Div full nomp>
        <P>Total</P>
        <P>{cost.toFixed(2)}$</P>
      </Div>
    </Div>
  );
}
