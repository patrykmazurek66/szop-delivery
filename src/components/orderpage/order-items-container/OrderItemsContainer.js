import OrderItem from "./order-item/OrderItem";
import { useState, useEffect } from "react";
import "./orderitemscontainer.css";

export default function OrderItemsContainer() {
  const [cart, setCart] = useState([]);
  const [items, setItems] = useState([]);
  const [cost, setCost] = useState(0);

  function refresh() {
    setCart(JSON.parse(localStorage.cart));
  }

  useEffect(() => {
    setCart(JSON.parse(localStorage.cart));
  }, []);

  useEffect(() => {
    setItems(
      cart.map(item => {
        return (
          <OrderItem
            name={item.name}
            price={item.price}
            qty={item.quantity}
            key={item.name}
            refresh={refresh}
          />
        );
      })
    );
    setCost(
      cart.reduce((acc, el) => {
        return (acc = acc + el.price * el.quantity);
      }, 0)
    );
  }, [cart]);

  return (
    <div className="order-items-container">
      <p className="order-items-container--header">Your Order</p>

      <div className="order-items-container--items">
        {items}
        <div className="order-items-container--items--delivery">
          <p>Delivery </p>5$
        </div>
      </div>
      <div className="order-items-container--total-cost">
        <p>Total</p>
        <p>{(cost + 5).toFixed(2)}$</p>
      </div>
    </div>
  );
}
