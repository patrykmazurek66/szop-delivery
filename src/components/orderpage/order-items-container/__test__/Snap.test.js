import renderer from "react-test-renderer";
import OrderItemsContainer from "../OrderItemsContainer";

it("Product renders correctly", () => {
  const tree = renderer.create(<OrderItemsContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});
