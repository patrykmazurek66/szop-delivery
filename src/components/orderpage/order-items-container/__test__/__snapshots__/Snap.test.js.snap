// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders correctly 1`] = `
<div
  className="order-items-container"
>
  <p
    className="order-items-container--header"
  >
    Your Order
  </p>
  <div
    className="order-items-container--items"
  >
    <div
      className="order-items-container--items--delivery"
    >
      <p>
        Delivery 
      </p>
      5$
    </div>
  </div>
  <div
    className="order-items-container--total-cost"
  >
    <p>
      Total
    </p>
    <p>
      5.00
      $
    </p>
  </div>
</div>
`;
